Feature: Create PSAT landing page and dashboard

mohdsayed opened the pull request:
## Description

Redesign the PSAT landing page and add a dashboard with pinned items and features for easy navigation.

## Relevant Technical Choices

- Remove the iframe and add static content.
- Map internal links for each item in the dashboard.
- Add internal link in PSAT landing page for dashboard.
- Display the landing page when users first install the PSAT extension. On subsequent visits, show the dashboard.
- Add top-level icons in collapsed sidebar.

## Testing Instructions

- Remove and install PSAT extension, you should be able to see the landing page.
- Open extension in a tab, you should now see dashboard.
- All links and buttons in the dashboard should navigate to the related pages.
- Top level icons should be visible in the collapsed sidebar.

## Additional Information:

<!-- Any other information. -->

## Screenshot/Screencast

Landing Page
<img width="1047" alt="image" src="https://github.com/user-attachments/assets/501d3bb9-c55a-4c25-98ba-cb7cc3e78be0">


Dashboard
<img width="1413" alt="image" src="https://github.com/user-attachments/assets/5ab80633-3c27-4833-8420-192645c7c6bb">


---

## Checklist

<!-- Check these after PR creation -->

- [x] I have thoroughly tested this code to the best of my abilities.
- [x] I have reviewed the code myself before requesting a review.
- ~[ ] This code is covered by unit tests to verify that it works as intended.~ NA
- [x] The QA of this PR is done by a member of the QA team (to be checked by QA).

<!--
Example:

Fixes #123
Partially addresses #22
See #834
-->

Fixes #799

mohdsayed commented:
@pavanpatil1 Ready for QA

pavanpatil1 commented:
- [ ] The button design should be consistent throughout.

<img width="1464" alt="Screenshot 2024-09-06 at 11 30 26 AM" src="https://github.com/user-attachments/assets/2476c2ce-5e2a-471a-b2d0-a62703fce27a">

---

- [ ] On the landing page, if CDP is enabled, the reload notice does not display correctly.

<img width="1463" alt="Screenshot 2024-09-06 at 11 46 52 AM" src="https://github.com/user-attachments/assets/dc391c79-d460-4d60-8d9f-a48c7cec65c6">

---


- [ ] Could we move the dashboard section above the cookie section? The dashboard typically appears first, and from there, we can navigate to the other sections below.

<img width="1452" alt="Screenshot 2024-09-06 at 12 18 04 PM" src="https://github.com/user-attachments/assets/22a42363-f4aa-42ab-ba89-8a8925ce6aa0">

---

- [ ] The color code on the dashboard does not match the one on the "Advertising" landing page. The landing page uses rgb(37, 99, 235), while the dashboard page uses rgb(26, 115, 232 )



<img width="1464" alt="Screenshot 2024-09-06 at 12 35 25 PM" src="https://github.com/user-attachments/assets/1c42cd95-7a6e-48aa-861b-b8b029bac57d">

---

milindmore22 commented:
- [ ] Disappearing logos 
The Privacy Sandbox logos on the PSAT landing page disappear when you toggle open the sidebar menus.

https://github.com/user-attachments/assets/8e4549a4-61f6-48d6-96de-0d03f9c84140

mohdsayed commented:
@pavanpatil1 

> The button design should be consistent throughout.

It's the same button used in dashboard. We wouldn't want to use the blue button.

>  On the landing page, if CDP is enabled, the reload notice does not display correctly.

Goo catch! Fixed

>  The logo currently appears in both the header and navigation sections. Could we streamline this by showing the logo in the header only? Also, the logo becomes invisible when navigating to the private audience or site boundary pages.

That makes sense, updated to use the same design from landing pages.

>  Could we move the dashboard section above the cookie section? The dashboard typically appears first, and from there, we can navigate to the other sections below.

I had initially kept it in the beginning; however, it was later decided that we wanted to keep it after “Privacy Sandbox”.

>  The color code on the dashboard does not match the one on the "Private Advertising" landing page. The landing page uses rgb(37, 99, 235), while the dashboard page uses rgb(26, 115, 232 )

Fixed

> The 'Private Advertising' logo differs between the landing page and the dashboard.

Fixed


@milindmore22 

>  Disappearing logos
The Privacy Sandbox logos on the PSAT landing page disappear when you toggle open the sidebar menus.

Fixed.

milindmore22 commented:
- [ ] The border for dashboard boxes is different from other box borders.

![image](https://github.com/user-attachments/assets/51238851-a923-4b33-94e4-91db51cbf69c)
![image](https://github.com/user-attachments/assets/2573ff2d-1799-4e48-8912-2993d1820bc8)

fellyph commented:
When the user clicks on the cookie table on the dashboard before running an audit, the link goes to the cookie landing page. Should we redirect the user to the cookie table after the user presses "analyses page"?
 

https://github.com/user-attachments/assets/c68144fd-7b21-4179-b4fb-c3d8b2de4784

mayan-000 commented on packages/extension/src/view/devtools/components/dashboard/constants.ts:
```diff
@@ -0,0 +1,133 @@
+    sidebarKey: SIDEBAR_ITEMS_KEYS.PROTECTED_AUDIENCE,
+  },
+  {
+    name: 'Cookies',
```
The translation is available for most names or descriptions, I18n can be used.

mayan-000 commented on packages/extension/src/view/devtools/components/dashboard/constants.ts:
```diff
@@ -0,0 +1,133 @@
+      },
+      {
+        name: 'Attribution',
+        sidebarKey: SIDEBAR_ITEMS_KEYS.ATTRIBUTION,
```
Wrong key, it's ATTRIBUTION_REPORTING

mohdsayed commented:
@fellyph 
> The dashboard should adapt to small screens and break the layout into two and one-column layout

Fixed

> When the user clicks on the cookie table on the dashboard before running an audit, the link goes to the cookie landing page. Should we redirect the user to the cookie table after the user presses "analyses page"?

I think instead of that, if the table is not available, that should not be displayed, pushed an update for it.
